### PR TITLE
Add ESLint rules to flag up unresolved paths

### LINKED ISSUE #471: Fix for errors stopping dev server running
image

I already have solved it in [this PR](https://github.com/SoftwareEngineeringDaily/sedaily-front-end/pull/472).
It's complaining about the paths used to access the file `utils/post.utils.js` but in the code, it's capitalised (i.e. `utils/Post.utils.js`).

image
> PostSummary.vue
image
> ForumThreadBody.vue
image

**Absolute** (or at least standardised aliases) trumps **relative** paths 👍 ⚡️

### PR DESCRIPTION
**Pull request by EmilyRosina**

### CODE REVIEW

File: src/components/PostSummary.vue [hunk 1]
```diff
@@ -135,7 +135,7 @@ export default {
 </script>
 
 <style scoped lang="stylus">
-@import './../css/variables'
+/* @import './../css/variables' */
```
Comment (jasonify) [inline src/components/PostSummary.vue]: Wondering what's going on here, is it just not being used or is it causing problems?

Comment (EmilyRosina) [reply src/components/PostSummary.vue]: Ah, my bad, i'll revert it now. It was my first stab at guessing what was wrong 😅

**Comment by EmilyRosina** [comment]
Is this okay to merge in now @jasonify @crablar?
Only, I still get the errors on any other branch I create 😅 unless I just merge this branch into any new branches first.

**Comment by tyler-furby** [comment]
@EmilyRosina Hey, I just reviewed this, what errors are you getting? From my end everything is looking good.

**Comment by EmilyRosina** [comment]
@tyler-furby same errors as previously mentioned in issue #471 - keep checking if someone else pushed a similar change but nada..

I cannot `npm run dev` or `npm run dev:api-staging`, as it cannot locate those files due to the capitalisation being wrong.

**Comment by jasonify** [comment]
Nice, seems like @andrewmarklloyd  and @joshmorel  ran into that bug here too: 

https://github.com/SoftwareEngineeringDaily/sedaily-front-end/pull/308

What's your dev environment btw?

**Comment by EmilyRosina** [comment]
@jasonify ah ok good, i'm not alone with this then 😅 

I'm on Windows with VS Code, but use linux (WSL) for my command line needs (git, node, npm etc).

**Comment by tyler-furby** [comment]
Do you notice a difference with `\` instead of `/`, windows is weird with that

**Comment by EmilyRosina** [comment]
No, that's invalid within paths in codebases, within the OS and command line it is that way though. 
The issue is that the file name is incorrectly cased, as raised in the issue Jason linked, so this PR fixes it 👍

**Comment by EmilyRosina** [comment]
Okay, but if anything, as mentioned in the other issue - travis etc will pick up on the bug that is naming a file incorrectly (as most servers run linux) - so if anything my setup makes more sense to keep as is surely?? I.e. we don't want WSL to ignore casing errors, as linux will through an error, as we won't have caught it...

Yeah, the PR should work, as it fixes the issue.. is that what you meant?

**Comment by jasonify** [comment]
😁  I think these badly name things are usually my fault since I'm on a mac. I've seen github have weird side-effects from this too.


I think this PR looks good to me & should be fine to merge in. Wondering if you guys already did a first quick pass at testing it (since it touches a few files)?

**Comment by jasonify** [comment]
I'm mostly on Atom but been using VS Code here and there.... though the main VIM plugin for it can be a little buggy for me so I still mostly stick to Atom :atom:

**Comment by EmilyRosina** [comment]
Even just a quick google, your [best option](https://github.com/atom-community/autocomplete-paths#defining-scopes) for atom isn't on par with [vs code's](https://marketplace.visualstudio.com/items?itemName=christian-kohler.path-intellisense) which is a real shame as it makes things so much easier, especially with the aliases setup in webpack already (just need to config the ext. to match).

**Comment by joshmorel** [comment]
I think a good choice to prevent this is the [eslint-plugin-import](https://github.com/benmosher/eslint-plugin-import) which includes a [no-unresolved rule](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-unresolved.md). Usage of the rule should help with IntelliSense across different OS/IDEs and if we use husky [like in our backend](https://github.com/SoftwareEngineeringDaily/software-engineering-daily-api/blob/develop/package.json) we can enforce this upon code commit.

eslint-import

@jasonify  if you agree I can put in PR.

**Comment by EmilyRosina** [comment]
Okay, so just tried adding as dependency with no config added (is that what you also did @joshmorel?)

Now getting both eslint _at top_ and original _bottom_ errors in the console - success!
image

**Comment by EmilyRosina** [comment]
Still unsure how i was getting the es-lint import errors without addign to config if all disabled by default BUT v interesting the amount of stuff that gets spewed into the console when i add `- eslint:recommended` to `extends:` in `.eslintrc.js` 😱

**Comment by EmilyRosina** [comment]
@joshmorel @jasonify @tyler-furby 
if you pull this branch and run on your local, do you get the error now?

**Comment by joshmorel** [comment]
Just taking a look now. Both of the above look like non-eslint errors. :slightly_smiling_face: Here's an example of an eslint error once the config is added:

maim1524781400

Looks like it was not quite as straight forward as I originally thought. We actually need a 2nd library because we are using a webpack resolver aliases `@`. But no problem, I got it working and made a PR against your own branch @EmilyRosina . Take [a look](https://github.com/EmilyRosina/sedaily-front-end/pull/1) and let me know if everything makes sense.

**Comment by EmilyRosina** [comment]
@joshmorel looking more at this today hopefully, but still not 100% happy with that error, it's misleading as it points to the '@' being wrong not the filename 😒 
Will delve further..

**Comment by joshmorel** [comment]
I wouldn't worry about that part. The error is pointing to the start of the path not the specific piece that is wrong (see alternate example below). Once you look at the import path and fix whatever part is wrong, the error will go away. The whole webpack alias does unfortunately add a complicating factor to reasoning about all this.

I did make a mistake in my commit, however. I had commented out the rule when examining different behaviour. This has now been resolved.

Another option to definitely consider is just to fix the import path case issues without adding the eslint library, and work on incorporating the library as a different PR. This may be a better approach since it's not super straight forward.

**Comment by EmilyRosina** [comment]
How do we feel about just enabling the whole plugin?
image

The only other error that arose was complaining about no default export for [filters](https://vuejs.org/v2/guide/filters.html)
image
I swapped some syntax and corrected import in tests, now no errors at all. Navigated through site and no ui errors popping up either.

**Comment by joshmorel** [comment]
If we extend the whole plugin, we'll need to disable certain rules such as this one, for example at the start of the file:

`/* eslint-disable import/default */`

The change you made may still work, but because it was already working and is unrelated, and especially because we have no automated tests, it's best just to leave it as is and add the disabling line.

**Pull request closed without merging**